### PR TITLE
Improve contributor docs

### PR DESCRIPTION
**Pull request by thomasvandoren**
* Add concise overview of steps to contributor info doc. This will be helpfulfor devs familiar with github and typical workflows.
* Add overview steps to CONTRIBUTING.md.
* Rename ContributorInfo.txt and GitCheatsheet.txt to use .rst suffix. They were already valid ReStructured Text files; this change makes it so github will render them.
  * [ContributorInfo.rst](https://github.com/thomasvandoren/chapel/blob/improve-contributor-docs/doc/developer/bestPractices/ContributorInfo.rst)
  * [GitCheatsheet](https://github.com/thomasvandoren/chapel/blob/improve-contributor-docs/doc/developer/bestPractices/GitCheatsheet.rst)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 in order to help us maintain Chapel's license terms.  The
-contributor's agreement can be found
-[here](https://github.com/chapel-lang/chapel/tree/master/doc/developer/contributorAgreements/).
+contributor's agreement can be found in [doc/developer/contributorsAgreements/][0].
 
-For more detailed instructions on committing code to Chapel, please
-refer to the contributor's
-[README](https://github.com/chapel-lang/chapel/tree/master/doc/developer/bestPractices/ContributorInfo.txt).
+For more detailed instructions on committing code to Chapel, please refer to
+[ContributorInfo.txt][1]. The high level steps are:
```
Comment (bradcray) [inline CONTRIBUTING.md]: I'd suggest a slight reorganization here for clarity.  Essentially:

The high-level steps for contributing code to Chapel are:

1 ...
2 ...
3 ...
4 ...
5 ...

For more detailed instructions, please refer to ...

**Comment by bradcray** [comment]
Can you remind me: Is ContributorInfo.txt in markdown or rst?  If the former, should we rename it so that it gets rendered properly when browsed within GitHub?

In either case, what's the best way for me to view the new copy of the file in rendered mode for review purposes?

File: doc/developer/bestPractices/ContributorInfo.txt [hunk 2]
```diff
@@ -13,6 +13,16 @@ account, signing the contributor agreement, and submitting pull requests.
...
 
+Overview:
+
+#. `Fork repo on github`_
+#. `Create new branch`_
+#. `Submit pull request`_
+#. `Test changes`_ (can be done by team at Cray)
+#. `Mail chapel-developers@ for review/merge`_
+
+.. _Fork repo on github:
```
Comment (bradcray) [inline doc/developer/bestPractices/ContributorInfo.txt]: Note that, whether intentionally or not, these anchors aren't rendered in the .html version of this file you pointed me to.  I think one would want to see the steps corresponding to the five items in the summary in the body of the documentation as well, in part to have a visual anchor to relate back to the quick summary; and in part to break up what is otherwise a pretty large file.

Main downside to doing so that I can see is that it makes reading the file as a .txt file a little less attractive.  But if we changed our www developer resources page to point to the file on GitHub the number of people who read it as a text file will go down significantly (and as all the docs/ content starts getting reflected at chapel.cray.com, it'll matter less and less).

Comment (thomasvandoren) [inline doc/developer/bestPractices/ContributorInfo.txt]: The links do work on github (fortunately). I'll add section titles.

Comment (bradcray) [inline doc/developer/bestPractices/ContributorInfo.txt]: I believe they work on HTML as well, it's just difficult to tell without the visual indicator that you've landed where you expect.

File: doc/developer/bestPractices/README [hunk 3]
```diff
@@ -9,7 +9,7 @@ developers.  A possible reading order is roughly as follows:
 GettingStarted.txt
   A list of suggested "first steps" for new developers.
```
Comment (bradcray) [inline doc/developer/bestPractices/README]: Is GitCheatSheet missing an entry here?  If so, would you add one while here?  (or else pull the GitCheatSheet renaming into a separate commit and do it there).

**Comment by bradcray** [comment]
This looks good to me.  We need a symmetric pull request in the internal repository for the www changes (ideally before this gets merged -- I point people to that first paragraph on almost a weekly basis now).
